rizinorg/rizin
# Fix case where chained fixups has less segments

**Your checklist for this pull request**
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why. Reason: no documentation present.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible) Reason: the binary is not redistributable.
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed) Reason: not needed.

**Detailed description**

If you install Kernel Debug Kit 12.2 build 21D49, you'll get:
`/Library/Developer/KDKs/KDK_12.2_21D49.kdk/System/Library/Kernels/kernel.development.t8101`

The file has 23 segments, but contains only 22 dyld chained fixups.

**Test plan**

1. Install Kernel Debug Kit 12.2 build 21D49. For details, see: https://developer.apple.com/documentation/apple-silicon/debugging-a-custom-kernel-extension
2. Open `/Library/Developer/KDKs/KDK_12.2_21D49.kdk/System/Library/Kernels/kernel.development.t8101` and confirm it does not crash.

**Closing issues**
